aml/src/resource.rs

```diff
@@ -1,112 +1,293 @@
+     *                                                        0  non-writeable (read-only))
+     * Byte 4     Range base address, _BAS bits [7:0]     Address bits [7:0] of the base memory address for which the card may be configured. 
+     * Byte 5     Range base address, _BASbits [15:8]     Address bits [15:8] of the base memory address for which the card may be configured.
+     * Byte 6     Range base address, _BASbits [23:16]    Address bits [23:16] of the base memory address for which the card may be configured. 
+     * Byte 7     Range base address, _BASbits [31:24]    Address bits [31:24] of the base memory address for which the card may be configured.
+     * Byte 8     Range length, _LEN bits [7:0]           This field contains bits [7:0] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     * Byte 9     Range length, _LEN bits[15:8]           This field contains bits [15:8] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     * Byte 10    Range length, _LEN bits [23:16]         This field contains bits [23:16] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     * Byte 11    Range length, _LEN bits [31:24]         This field contains bits [31:24] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     */
+    if bytes.len() < 12 {
+        return Err(AmlError::ResourceDescriptorTooShort);
+    }
+
+    let information = bytes[3];
+    let is_writable = information.get_bit(0);
+
+    let base_address = 
+        ((bytes[4] as u32) << 0) | 
+        ((bytes[5] as u32) << 8) | 
+        ((bytes[6] as u32) << 16) | 
+        ((bytes[7] as u32) << 24);
+
+    let range_length = 
+        ((bytes[8] as u32) << 0) | 
+        ((bytes[9] as u32) << 8) | 
+        ((bytes[10] as u32) << 16) | 
+        ((bytes[11] as u32) << 24);
+
+    Ok(Resource::MemoryRange(MemoryRangeDescriptor::FixedLocation {
+        is_writable,
+        base_address,
+        range_length
+    }))
+}
+
+fn address_space_descriptor<T>(bytes: &[u8]) -> Result<Resource, AmlError> {
+    if bytes.len() < 14 {
+        return Err(AmlError::ResourceDescriptorTooShort);
+    }
+
+    let resource_type = match bytes[3] {
+        0 => AddressSpaceResourceType::MemoryRange,
+        1 => AddressSpaceResourceType::IORange,
+        2 => AddressSpaceResourceType::BusNumberRange,
+        3..=191 => return Err(AmlError::ReservedResourceType),
+        192..=255 => unimplemented!()
+    };
+
+    let general_flags = bytes[4];
+    let is_maximum_address_fixed = general_flags.get_bit(3);
+    let is_minimum_address_fixed = general_flags.get_bit(2);
+    let decode_type = if general_flags.get_bit(1) {
+        AddressSpaceDecodeType::Subtractive
+    } else {
+        AddressSpaceDecodeType::Additive
+    };
+
+    let size = core::mem::size_of::<T>();
```

Comment:
Tiny nit (feel free to ignore): I like to bring `core::mem` into scope in the whole file and use `mem::size_of`